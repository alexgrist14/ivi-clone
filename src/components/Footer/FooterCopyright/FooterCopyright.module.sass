.copyright
  margin-block: 32px 48px
  font-size: 15px
  line-height: 20px
  color: rgba(255,255,255,.48)

@media (0 < width < 1159px)
  .copyright
    font-size: 13px
    line-height: 16px
    color: #312b45
