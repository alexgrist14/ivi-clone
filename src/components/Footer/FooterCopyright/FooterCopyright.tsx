import { FC } from "react";
import styles from './FooterCopyright.module.sass';

const today = new Date();

const FooterCopyright: FC = () => {
  return (
    <aside className={styles.copyright}>
      <span>
        © {today.getFullYear()} ООО «Иви.ру»
        <br /> HBO ® and related service marks are the property of Home Box
        Office, Inc
      </span>
    </aside>
  );
};

export default FooterCopyright;
