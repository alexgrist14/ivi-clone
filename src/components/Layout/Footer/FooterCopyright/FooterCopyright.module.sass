@use '/src/styles/vars'

.copyright
  margin-block: 32px 48px
  font-size: 15px
  line-height: 20px
  color: vars.$primaryColor

@media (0 < width < 1159px)
  .copyright
    font-size: 13px
    line-height: 16px
    color: #312b45
