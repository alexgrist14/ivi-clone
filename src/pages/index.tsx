import Head from 'next/head'
import Banner from '../components/Banner/Banner'
import Footer from '../components/Footer/Footer';
import Header from '/src/components/Header/Header';

export default function Home() {
  return (
    <>
      <Head>
        <title>Иви</title>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <main>
        <Header />
        <Banner />
        <Footer />
      </main>
    </>
  )
}
