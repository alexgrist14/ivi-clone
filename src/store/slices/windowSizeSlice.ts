import { createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import IWindowSize from "/src/types/IWindowSize";

const initialState: IWindowSize = { width: 0, height: 0 };

export const windowSizeSlice = createSlice({
  name: "windowSize",
  initialState,
  reducers: {
    setWindowSize: (state, action: PayloadAction<IWindowSize>) => {
      state.width = action.payload.width;
      state.height = action.payload.height;
    },
  },
});

export const { setWindowSize } = windowSizeSlice.actions;
export default windowSizeSlice.reducer;
