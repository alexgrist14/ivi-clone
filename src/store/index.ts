import { configureStore } from "@reduxjs/toolkit";
import windowSizeSlice from "./slices/windowSizeSlice";
import modalsSlice from "./slices/modalsSlice";
import slugsSlice from "/src/store/slices/slugsSlice";
import filtersSlice from "./slices/filtersSlice";
import authSlice from "/src/store/slices/authSlice";

export const reducersSetup = {
  windowSize: windowSizeSlice,
  showModal: modalsSlice,
  slugs: slugsSlice,
  filters: filtersSlice,
  auth: authSlice,
};

export const store = configureStore({
  reducer: reducersSetup,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
