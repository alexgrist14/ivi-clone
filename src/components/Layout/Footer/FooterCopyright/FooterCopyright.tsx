import { FC } from "react";
import { useTranslation } from "next-i18next";
import styles from "./FooterCopyright.module.sass";
import { useAppSelector } from "/src/hooks/redux";
import { RootState } from "/src/store";

const today = new Date();

const FooterCopyright: FC = () => {
  const windowSize = useAppSelector((state: RootState) => state.windowSize);
  const { t } = useTranslation("footer");

  return (
    <aside data-testid="copyright" className={styles.copyright}>
      <span>
        {windowSize.width > 1159 && (
          <span>
            © {today.getFullYear()} ООО «{t("copyright")}»
            <br />
          </span>
        )}
        HBO ® and related service marks are the property of Home Box Office, Inc
      </span>
    </aside>
  );
};

export default FooterCopyright;
