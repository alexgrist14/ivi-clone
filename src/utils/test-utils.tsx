import { ReactNode } from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { reducersSetup } from "/src/store";
import { configureStore } from "@reduxjs/toolkit";
import { IProviderRender } from "../types/IRender";

export const renderWithProviders = (children: ReactNode): IProviderRender => {
  const store = configureStore({ reducer: reducersSetup });
  const component = render(<Provider store={store}>{children}</Provider>);

  return { store, component };
};
