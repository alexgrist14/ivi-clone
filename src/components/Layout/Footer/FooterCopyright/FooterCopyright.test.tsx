import FooterCopyright from "./FooterCopyright";
import { renderWithProviders } from "/src/utils/test-utils";
import { act, screen } from "@testing-library/react";
import { store } from "/src/store";
import { setWindowSize } from "/src/store/slices/windowSizeSlice";

describe("Footer copyright", () => {
  beforeEach(() => {
    renderWithProviders(<FooterCopyright />);
  });
  test("should renders without errors", () => {
    expect(screen.getByTestId("copyright")).toBeInTheDocument();
  });

  test("render with width more than 1159px", () => {
    act(() => store.dispatch(setWindowSize({ width: 1920, height: 1080 })));
    expect(screen.getByTestId("copyright")).toMatchSnapshot();
  });

  test("render with width less than 1160px", () => {
    act(() => store.dispatch(setWindowSize({ width: 1159, height: 1080 })));
    expect(screen.getByTestId("copyright")).toMatchSnapshot();
  });
});
