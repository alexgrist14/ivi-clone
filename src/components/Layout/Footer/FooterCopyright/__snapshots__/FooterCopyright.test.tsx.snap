// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer copyright render with width less than 1160px 1`] = `
<aside
  class="copyright"
  data-testid="copyright"
>
  <span>
    HBO ® and related service marks are the property of Home Box Office, Inc
  </span>
</aside>
`;

exports[`Footer copyright render with width more than 1159px 1`] = `
<aside
  class="copyright"
  data-testid="copyright"
>
  <span>
    HBO ® and related service marks are the property of Home Box Office, Inc
  </span>
</aside>
`;
